pfioh/store-options-handler.py

```diff
@@ -0,0 +1,322 @@
+            self.wfile.write(fh.read())
+        
+        #Cleanup by deleting temporary files
+        if b_cleanup:
+            if b_zip:
+                self.qprint("Removing '%s'..." % (str_zipFile), comms = 'status')
+                if os.path.isfile(str_zipFile):     os.remove(str_zipFile)
+            if str_encoding == 'base64':
+                self.qprint("Removing '%s'..." % (str_base64File), comms = 'status')
+                if os.path.isfile(str_base64File):  os.remove(str_base64File)
+
+        return d_ret
+
+
+class SwiftStore(StoreHandler):
+
+	def __init__(self, *args, **kwargs):
+		StoreHandler.__init__(self, *args, **kwargs)
+
+		"""
+		NOTE:
+		Default Paramters to be set by Openshift:
+		osAuthUrl,username,password,osProjectDomain,osProjectName,containerName
+		"""
+
+
+	def _getScopedSession(self, osAuthUrl, username, password, osProjectDomain, osProjectName):
+    	"""
+    	Uses keystone authentication to create and return a scoped session
+    	"""
+    	
+    	passwordAuth = v3.Password(auth_url= osAuthUrl,
+                                user_domain_name= 'default',
+                                username= username, password= password,
+                                project_domain_name= osProjectDomain,
+                                project_name= osProjectName,
+                                unscoped= False)
+
+    	scopedSession = session.Session(auth= passwordAuth)
+    	return scopedSession
+
+
+	def _initiateSwiftConnection(self, **kwargs):
+		"""
+		Initiates a Swift connection and returns a Swift Connection Object
+		"""
+
+		#Paramters will be initialized by template
+
+		scopedSession = self._getScopedSession(osAuthUrl, username, password, osProjectDomain, osProjectName)
+		swiftConnection = swift_client.Connection(session=scopedSession)
+		return swiftConnection
+
+
+	def _putObject(self, swiftConnection, containerName, key, value):
+		"""
+		Creates an object with the given key and value and puts the object in the specified container
+		"""
+
+		try:
+			swiftConnection.put_object(containerName, key , contents=value, content_type='text/plain')
+			self.qprint('Object added with key %s' %key)
+
+		except Exception as exp:
+			self.qprint(exp)
+
+
+	def _getObject(self, swiftConnection, containerName, key, b_Delete):
+		"""
+		Returns an object associated with the specified key in the specified container
+		Deletes the object after returning if specified
+		"""
+
+		try:
+			swiftDataObject = swiftConnection.get_object(containerName, key)
+
+			if b_Delete:
+				swiftConnection.delete_object(containerName, key)
+				self.qprint('Deleted object with key %s' %key)
+
+		except Exception as exp:
+			self.qprint(exp)
+
+		return swiftDataObject
+
+
+	def storeData(self, **kwargs):
+		"""
+		Creates an object of the file and stores it into the container as key-value object 
+		"""
+
+		for k,v in kwargs.items():
+			if k == 'file_content':   fileContent = v
+            if k == 'Path': 	  key= v

```

Comment:
Fixing it. Strangely, sublime text shows proper indentation, however when I open it in gedit the indentation is messed up.